doc/monads/st.lean

```diff
@@ -0,0 +1,117 @@
+/-!
+# ST monad
```

Comment:
```suggestion
# References and the ST monad
```
It makes sense to explain IO.Ref in the same section, and I would try to use this section to motivate the concept of references and explain ST toward the end as a way to use references in pure code.

Reply:
well the problem is this is the chapter on "Monads" not the chapter on "References".  It seemed to me that using the ST monad implies using ST.Ref ?  Or am I conflating 2 different things here?